### PR TITLE
[UI] - RTK Migration Designs

### LINKED ISSUE #10046: [UI] - RTK Migration Designs
### Desired Behavior

All the api in the "Designs" page migrate to RTK.

- [ ] Publish design - `api/pattern/catalog/publish` [POST] -> https://docs.meshery.io/reference/rest-apis#api-pattern-catalog-publish @cesartheroman
- [ ] Unpublish design - `api/pattern/catalog/unpublish` [DELETE] -> https://docs.meshery.io/reference/rest-apis#api-pattern-catalog-unpublish @cesartheroman
- [ ] Deploy design - `api/pattern/deploy` [POST] -> https://docs.meshery.io/reference/rest-apis#api-pattern-deploy
- [ ] Un deploy design - `api/pattern/deploy` [DELETE] -> https://docs.meshery.io/reference/rest-apis#api-pattern-deploy

https://playground.meshery.io/configuration/designs

### Mockups

Get some idea how to do migration - https://github.com/meshery/meshery/issues/9985

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by cesartheroman**
**Notes for Reviewers**

This PR fixes #10046 and related #9985
@senali-d @sudhanshutech 

- [X] publish design 
   - api/pattern/catalog/publish [POST] -> https://docs.meshery.io/reference/rest-apis#api-pattern-catalog-publish 
- [X] Unpublish design 
  - api/pattern/catalog/unpublish [DELETE] -> https://docs.meshery.io/reference/rest-apis#api-pattern-catalog-unpublish
- [X] Deploy design 
   - api/pattern/deploy [POST] -> https://docs.meshery.io/reference/rest-apis#api-pattern-deploy
- [X] Undeploy design 
   - api/pattern/deploy [DELETE] -> https://docs.meshery.io/reference/rest-apis#api-pattern-deploy

I left a note for the deploy and undeploy calls since i had to hardcode the `/api/pattern/deploy` because I was getting an extra `/api` prefixed through `ctxUrl` for some reason. This behavior didn't happen in the original `dataFetch` implementation so I suspect `rtk-query` might be to blame here but I'm not sure. 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by cesartheroman** [comment]
@sudhanshutech look good?

File: ui/components/MesheryPatterns.js [hunk 1]
```diff
@@ -765,29 +766,54 @@ function MesheryPatterns({
...
+            .catch((err) => {
+              notify({
+                message: `${ACTION_TYPES.UNPUBLISH_CATALOG.error_msg}: ${err.error}`,
+                event_type: EVENT_TYPES.ERROR,
+                details: err.toString(),
+              });
+            });
         }
+
+        //Previous implementation
```
Comment (sudhanshutech) [inline ui/components/MesheryPatterns.js:804]: you can remove this

File: ui/components/MesheryPatterns.js [hunk 2]
```diff
@@ -908,9 +938,36 @@ function MesheryPatterns({
...
+      })
+      .catch((err) => {
+        notify({
+          message: `${ACTION_TYPES.PUBLISH_CATALOG.error_msg}: ${err.error}`,
+          event_type: EVENT_TYPES.ERROR,
+          details: err.toString(),
+        });
+      });
+
+    //Previous implementation:
```
Comment (sudhanshutech) [inline ui/components/MesheryPatterns.js:950]: remove

**Comment by sudhanshutech** [comment]
looks good, are you also migrating other area to rtk here in designs @cesartheroman ?

File: ui/rtk-query/design.js [hunk 3]
```diff
@@ -23,7 +23,27 @@ const designs = api
         }),
         providesTags: () => [{ type: TAGS.DESIGNS }],
       }),
+
+      addPatterns: builder.mutation({
```
Comment (sudhanshutech) [inline ui/rtk-query/design.js:27]: publishPattern is more appropriate naming convention

File: ui/rtk-query/design.js [hunk 4]
```diff
@@ -23,7 +23,27 @@ const designs = api
...
+        query: (queryArg) => ({
+          url: `pattern/catalog/publish`,
+          credentials: 'include',
+          method: 'POST',
+          body: queryArg.patternPayload,
+        }),
+        //provides or invalidates any tags?
+      }),
+
+      removePatterns: builder.mutation({
```
Comment (sudhanshutech) [inline ui/rtk-query/design.js:37]: same unpublishPattern here

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

**Pull request closed without merging**